pymc_marketing/mmm/multidimensional.py

```diff
@@ -0,0 +1,1653 @@
+                # Select Y data (posterior contributions)
+                y_data = self.idata.posterior.channel_contribution.sel(**indexers)
+
+                # Flatten chain & draw by taking mean (or sum, up to design)
+                y_data = y_data.mean(dim=["chain", "draw"])
+
+                # Ensure X and Y have matching date coords
+                x_data = x_data.broadcast_like(y_data)
+                y_data = y_data.broadcast_like(x_data)
+
+                # Scatter
+                ax.scatter(
+                    x_data.values.flatten(),
+                    y_data.values.flatten(),
+                    alpha=0.8,
+                    color=f"C{row_idx}",
+                )
+
+                title = self._build_subplot_title(
+                    dims=["channel", *additional_dims],
+                    combo=(channel, *combo),
+                    fallback_title="Channel Saturation Curves",
+                )
+                ax.set_title(title)
+                ax.set_xlabel("Channel Data (X)")
+                ax.set_ylabel("Channel Contributions (Y)")
+
+        return fig, axes
+
+
+class MMM(ModelBuilder):
+    """Media Mix Model class for estimating the impact of marketing channels on a target variable.
+
+    This class implements the core functionality of a Media Mix Model (MMM), allowing for the
+    specification of various marketing channels, adstock transformations, saturation effects,
+    and time-varying parameters. It provides methods for fitting the model to data, making
+    predictions, and visualizing the results.
+
+    Attributes
+    ----------
+    date_column : str
+        The name of the column representing the date in the dataset.
+    channel_columns : list[str]
+        A list of columns representing the marketing channels.
+    target_column : str
+        The name of the column representing the target variable to be predicted.
+    adstock : AdstockTransformation
+        The adstock transformation to apply to the channel data.
+    saturation : SaturationTransformation
+        The saturation transformation to apply to the channel data.
+    time_varying_intercept : bool
+        Whether to use a time-varying intercept in the model.
+    time_varying_media : bool
+        Whether to use time-varying effects for media channels.
+    dims : tuple | None
+        Additional dimensions for the model.
+    model_config : dict | None
+        Configuration settings for the model.
+    sampler_config : dict | None
+        Configuration settings for the sampler.
+    control_columns : list[str] | None
+        A list of control variables to include in the model.
+    yearly_seasonality : int | None
+        The number of yearly seasonalities to include in the model.
+    adstock_first : bool
+        Whether to apply adstock transformations before saturation.
+    """
+
+    _model_name: str = "BaseMMM"
+    _model_type: str = "BaseValidateMMM"
+    version: str = "0.0.1"
```

Comment:
sure about these names? they do not mention anything "multidimensional"

Reply:
Changed. How is MMMM? 😄